gpii/node_modules/flowManager/test/SolutionsRegistryDataSourceTests.js

```diff
@@ -0,0 +1,320 @@
+            options: {
+                path: gpii.tests.solutionsRegistry.path
+            }
+        }
+    },
+    modules: [{
+        name: "Solutions registry used by CBFM - file-based solutions",
+        expect: 2,
+        tests: [{
+            name: "Solutions registry used with CBFM",
+            sequence: [
+                { funcName: "fluid.log", args: ["Solutions Registry for CBFM, getting named registry -- START"] },
+
+                // For some reason, even though the solutionsRegistryReady event
+                // fires, this doesn't hear it.
+                // ... and doesn't appear to need to wait for it (?!)
+                /*{
+                    event: "{solutionsRegistryDataSource}.events.solutionsRegistryReady",
+                    listener: "jqUnit.assert",
+                    args: ["CBFM solutionsRegistryDataSource ready"]
+                },*/ {
+                    funcName: "{solutionsRegistryDataSource}.get",
+                    args: [gpii.tests.solutionsRegistry.requestOptions]
+                }, {
+                    event: "{solutionsRegistryDataSource}.events.onRead",
+                    listener: "gpii.tests.solutionsRegistry.checkOneRegistry",
+                    args: [
+                        "Expecting one solutions registry",
+                        gpii.tests.solutionsRegistry.requestOptions.os,
+                        "{arguments}.0"
+                    ]
+                },
+                { funcName: "fluid.log", args: ["Solutions Registry for CBFM, getting named registry -- END"] }
+            ]
+        }]
+    }]
+});
+
+// Test environment for CLFM solutions registry data source
+fluid.defaults("gpii.tests.solutionsRegistry.dataSource.cloud.env", {
+    gradeNames: ["fluid.test.testEnvironment"],
+    components: {
+        tester: {
+            type: "gpii.tests.solutionsRegistry.cloud.testCaseHolder"
+        }
+    }
+});
+kettle.test.bootstrap("gpii.tests.solutionsRegistry.dataSource.cloud.env");
+
+// ======== Testing the solutions registry data source used by the LFM ========
+
+// Test component for the LFM solutions registry data source mocking a load
+// from the source code repository.
+fluid.defaults("gpii.tests.solutionsRegistry.dataSource.local.mockLoadFromRepository", {
+    gradeNames: ["gpii.flowManager.solutionsRegistry.dataSource.local"],
+    components: {
+        revisionRequester: {
+            type: "gpii.flowmanager.revisionRequester",
+            options: {
+                cloudURL: "http://gpii.net"
+            }
+        }
+    },
+    listeners: {
+        "loadSolutions.loadFromLocalDisk": {
+            listener: "{that}.loadFromLocalDisk",
+            priority: "first"
+        },
+        "loadSolutions.getRevision": {
+            listener: "{revisionRequester}.getRevision",
+            priority: "after:loadFromLocalDisk"
+        },
+        "loadSolutions.loadFromRepository": {
```

Comment:
This listener appears the only one required for overriding the base component. Please consider to remove other 3 listeners so when they are modified in the base component. These 3 here won't unintentionally override ones from the base.

Reply:
I tried that initially, and nothing worked.  Then, I tried different combinations of the listeners until I ended up replicating the entire set from the base grade.  However, I've since tried your suggestion, and it's fine.  It must have been a typo, before, I guess.  Upshot:  removed all but the critical listener.